### ISSUE TITLE
Parallel pylint run inflates "Messages" report occurrences while emitted diagnostics stay unchanged

### ISSUE THREAD

**Issue by aleksandr-kotlyar** [PyCQA/pylint#10996]
### Bug description

```python
When running pylint with multiple jobs, the final `Messages` report table can show a heavily inflated number of occurrences, while the actual emitted diagnostics stay unchanged.

In other words, the lint messages printed in the log are correct, but the aggregated report statistics at the bottom are not.

This is especially confusing in CI, because the report table suggests that the project has an order-of-magnitude larger number of violations than the actual lint output contains.

### Observed behavior

On a real project, using the same input file list and the same pylint config:


pylint --rcfile=.linters/.pylintrc --persistent=no --jobs=1 $files > pylint-j1.log || true
pylint --rcfile=.linters/.pylintrc --persistent=no --jobs=4 $files > pylint-j4.log || true

The emitted missing-function-docstring diagnostics are identical:

--jobs=1:
  emitted C0116 / missing-function-docstring lines: 2084
  Messages table occurrence count: 2084

--jobs=4:
  emitted C0116 / missing-function-docstring lines: 2084
  Messages table occurrence count: 131819

I also verified that:

- the input does not include `.venv`;
- the input does not include `site-packages`;
- the input is an explicit list of tracked project `.py` files;
- `--persistent=no` does not fix the inflated report count;
- switching to `--jobs=`1 fixes the report count.

So this does not appear to be caused by pylint-cache persistence or by accidentally linting dependencies.
```

### Configuration

```ini
[MAIN]
jobs=4
persistent=yes

[REPORTS]
reports=yes
score=yes

The issue is triggered by parallel execution (`jobs=4` / `--jobs=4`) when reports are enabled.

`persistent=yes` is present in the project configuration, but the inflated `Messages` report is still reproduced when pylint is run with `--persistent=no`, so persistence does not appear to be the root cause.
```

### Command used

```shell
## Possible minimal reproducer
tmpdir="$(mktemp -d)"
cd "$tmpdir" || exit 1

python -m venv .venv
. .venv/bin/activate

pip install "pylint==4.0.1"

for i in $(seq 1 80); do
  cat > "module_${i}.py" <<'PY'
def first_function():
    return 1

def second_function():
    return 2
PY
done

pylint \
  --reports=y \
  --persistent=no \
  --jobs=1 \
  --disable=all \
  --enable=missing-function-docstring \
  module_*.py > pylint-j1.log || true

pylint \
  --reports=y \
  --persistent=no \
  --jobs=4 \
  --disable=all \
  --enable=missing-function-docstring \
  module_*.py > pylint-j4.log || true

echo "emitted C0116 with --jobs=1:"
grep -c 'C0116:' pylint-j1.log || true

echo "emitted C0116 with --jobs=4:"
grep -c 'C0116:' pylint-j4.log || true

echo "Messages table with --jobs=1:"
grep -A20 -E 'Messages|message id' pylint-j1.log || true

echo "Messages table with --jobs=4:"
grep -A20 -E 'Messages|message id' pylint-j4.log || true
```

### Pylint output

```python
--jobs=4:

+----------------------------+-------------+
|message id                  |occurrences  |
+============================+=============+
|missing-function-docstring  |131819       |
+----------------------------+-------------+

--jobs=1:

+----------------------------+-------------+
|message id                  |occurrences  |
+============================+=============+
|missing-function-docstring  |2084         |
+----------------------------+-------------+
```

### Expected behavior

The occurrence count in the final `Messages` report should match the actual number of emitted diagnostics, regardless of the value of `--jobs`.

For example, if pylint emits 2084 `C0116: missing-function-docstring` diagnostics, the final report should show:
```
missing-function-docstring    2084
```
both with `--jobs=1` and with `--jobs=4`.

## Actual behavior
With `--jobs=4`, pylint emits 2084 `C0116` diagnostics, but the final Messages report shows a much larger number, for example:
```
missing-function-docstring    131819
```

### Pylint version

```shell
pylint 4.0.5
astroid 4.0.4
Python 3.12.7 (main, Oct  1 2024, 02:05:46) [Clang 15.0.0 (clang-1500.3.9.4)]
```

### OS / Environment

OS: Linux CI runner
Python: 3.12
pylint: 4.0.1
astroid: 4.0.1
Shell: bash/sh
Execution context: GitLab CI

The issue is reproduced when pylint is run with `--jobs=4`.
The input is an explicit list of tracked project Python files.
`.venv` and `site-packages` are not included.
Using `--jobs=1` makes the final `Messages` report count match the emitted diagnostics.

### Additional dependencies

```python
Not applicable at the moment. The issue seems reproducible with a minimal pylint-only environment. I can provide the full `pip freeze` output if needed.
```
